src/runtime/virtcontainers/pkg/oci/utils.go

```diff
@@ -1166,3 +1042,53 @@ func IsCRIOContainerManager(spec *specs.Spec) bool {
 	}
 	return false
 }
+
+const (
+	errAnnotationPositiveNumericKey = "Error parsing annotation for %s: Please specify positive numeric value"
+	errAnnotationBoolKey            = "Error parsing annotation for %s: Please specify boolean value 'true|false'"
+)
+
+func parseAnnotationUint32Configuration(ocispec specs.Spec, key string, field *uint32, validFunc func(uint32) error) error {
```

Comment:
I'm wondering why pass a different validFunc and setFun for  parseAnnotationUint32Configuration and parseAnnotationUint64Configuration? Can we merge those two parse functions into one and just return the parsed value as uinit64 directly and let the caller to do the converting?